File: p006.py
def p006(n):
    return pow(sum(i for i in xrange(1, n + 1)), 2) - sum(pow(i, 2) for i in xrange(1, n + 1))

if __name__ == "__main__":
    from benchmark import benchmark
    benchmark(p006, 100)

File: p006.clj
(use '[euler :only [run]])

(defn p006 [n]
  (let [r (range (+ n 1))]
    (long (- (Math/pow (reduce + r) 2)
          (reduce + (map #(Math/pow % 2) r))))))

(run (p006 100))
